docs(operating modes): Add operating modes page (#2291)
stryker-mutator/stryker-net: docs/operating-modes.md

---
title: Operating modes
sidebar_position: 25
custom_edit_url: https://github.com/stryker-mutator/stryker-net/edit/master/docs/operating-modes.md
---
# Operating modes

> There are currently two different ways to run Stryker:

Reviewer: ```suggestion
There are currently two ways to run Stryker:
```

- From the test project context
- From the source project context

> 

Reviewer: Wait... you can run Stryker from wherever, if you specify both source and test projects, at least it works with full paths.

Reviewer 2: Wait does that work?... I never tried it but I would expect stryker to fail if you run it from a location outside of your project

Reviewer: It absolutely does :)

Reviewer: Well it's been there for a while, we have 1.5.3 in our pipeline currently and we are running things like this. 
So I don't know why and since when it works but it does... might be worth checking that the new custom output option is okay with that and also adding some integration tests then.

Author: I see, I don't know if it's a good idea to add this to the documentation as it's not really officially supported right now. It can be added later when this feature is properly tested.

---
title: Operating modes
sidebar_position: 25
custom_edit_url: https://github.com/stryker-mutator/stryker-net/edit/master/docs/operating-modes.md
---
# Operating modes

There are currently two different ways to run Stryker:

- From the test project context

> - From the target project context

Reviewer 3: ```suggestion
- From the source project context
```

@richardwerkman what do we want to call this? We've called it target project for a long time but that really only makes sense from the context of the test project. It doesn't really make sense anymore now that we can run stryker from different contexts.

Reviewer 2: Source makes more sense now for this. And should we call to solution one "source solution context"? Once it's done of course

You will also be able to run Stryker from your solution context in the future.

## Test project context

> Just run
> 
> ```bash
> dotnet stryker
> ```
> 
> in your test project folder. Stryker will automatically detect the target project based on the project reference. If your test project has multiple project references, you need to specify the target project with `--project <filename>` ([docs here](https://stryker-mutator.io/docs/stryker-net/configuration/#project-file-name)). You can only specify one target project at the time.

Reviewer 3: ```suggestion
Just run `dotnet stryker` in your test project folder.  
Stryker will automatically detect the target project based on the project reference. If your test project has multiple project references, you need to specify the target project with `--project <filename>` ([docs here](https://stryker-mutator.io/docs/stryker-net/configuration/#project-file-name)). You can only specify one target project at the time.
```

Author: Hier ook source project noemen dan?

```suggestion
Just run `dotnet stryker` in your test project folder.  
Stryker will automatically detect the source project based on the project reference. If your test project has multiple project references, you need to specify the source project with `--project <filename>` ([docs here](https://stryker-mutator.io/docs/stryker-net/configuration/#project-file-name)). You can only specify one source project at the time.
```

Reviewer 3: If @richardwerkman agrees with that naming

## Target project context

Run

```bash
dotnet stryker -tp "../Tests/Tests.csproj"
```

> in your target project folder. If your target project uses multiple test projects that you want to include in the Stryker run, you need to specify each test project. You can include multiple test projects like this ([see docs](https://stryker-mutator.io/docs/stryker-net/configuration/#test-projects-string)):

Reviewer 3: ```suggestion
in your target project folder. If multiple test projects (ex unit tests, specs) target the same source project you need to specify each test project. You can include multiple test projects like this ([see docs](https://stryker-mutator.io/docs/stryker-net/configuration/#test-projects-string)):
```

> in your target project folder. If your target project uses multiple test projects that you want to include in the Stryker run, you need to specify each test project. You can include multiple test projects like this ([see docs](https://stryker-mutator.io/docs/stryker-net/configuration/#test-projects-string)):

Author: Run `dotnet stryker -tp "../Tests/Tests.csproj"` in your target project folder.  
If your source project uses multiple test projects that you want to include in the Stryker run, you need to specify each test project. You can include multiple test projects like this ([see docs](https://stryker-mutator.io/docs/stryker-net/configuration/#test-projects-string)):
